[Alyssa Savier] iP

alyssa-savier opened the pull request:
Joker

> “What doesn't kill you simply makes you STRANGER.” – Joker ([source](https://www.youtube.com/watch?v=o5o-w0F7kfQ))

Joker frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2122S1/ip/pull/368).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

alyssa-savier committed:
Implement skeletal version of Duke with echo function
Duke renamed to Joker.

Joker greets the user, echos commands entered by user, and exits when user types bye.

alyssa-savier committed:
Add ability to store text entered by user and list back out for user

alyssa-savier committed:
Add the ability to mark tasks as done.
Creating a task class makes it much easier to mark tasks as done.
Another tasklist array also makes it easier to keep track.

alyssa-savier committed:
Add support for tracking three types of tasks: TodDos, Events, Deadlines
Created three classes ToDo, Event, Deadline, which inherit from the Task class.

alyssa-savier committed:
Add Automated Text UI Testing

alyssa-savier committed:
Add Error Handling for 2 types of errors
Error 1: Description of a todo list cannot be empty
Error 2: Invalid command/input

alyssa-savier committed:
Fix some bugs related to bye command

alyssa-savier committed:
Add support for deleting tasks from the list

alyssa-savier committed:
Refactor the Code
More OOP added.

alyssa-savier committed:
Refactor the Code
Added more OOP.

alyssa-savier committed:
Merge branch 'A-MoreOOP'

alyssa-savier committed:
Add automatically save tasks in hard disk feature
Automatically saves when task list changes. Loads data from hard disk
when Joker starts up.

alyssa-savier committed:
Add Support for Joker to read dates and times
Store deadline and event dates as a java.time.LocalDate. Accept dates in
this format: yyyy-mm-dd (e.g., 2019-10-15) and print in a different
format such as MMM dd yyyy (e.g., Oct 15 2019).

wpinrui commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,27 @@
+        }
+    }
+
+    public static void introMessage() {
```
Would it be clearer if this was named `printIntroMessage`?

wpinrui commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,289 @@
+import java.io.*;
```
Would it be possible to list all imports?

wpinrui commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,289 @@
+
+public class TaskManager {
+    Task[] taskList = new Task[100];
+    int i = 0;
```
Wonder if a more descriptive name would be helpful here?

wpinrui commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,289 @@
+public class TaskManager {
+    Task[] taskList = new Task[100];
+    int i = 0;
+    File txtFile;
```
Would it be better to avoid abbreviations and name this as `textFile`?

wpinrui commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,289 @@
+        } else {
+            try {
+                BufferedReader br = new BufferedReader(new FileReader(txtFile));
+                String txtLine;
```
Similarly, should this be named as `textLine`?

wpinrui commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,289 @@
+                }
+            } else if (inData.contains("done ")) {
+                if (isNumeric(inData.substring(5, inDataLength))) {
+                    int taskNo = Integer.parseInt(inData.substring(5, inDataLength));
```
Would it be better to call this `taskNumber` to prevent confusion with the negative reply 'No'?

wpinrui commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,289 @@
+        System.out.println("____________________________________________________________");
+    }
+
+    public void toDoAddedMessage() {
```
Would it be better to use an action to name this, like `printToDoAddedMessage` (similar to the other printing methods too)?

wpinrui reviewed (commented):
I think the naming is generally pretty good, save for a few tips that we have just learnt about naming methods with actions. Coding standard looks good too.

Anonymxtrix commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,26 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
```
Should there be header comments following the coding standards layed out [here](https://se-education.org/guides/conventions/java/intermediate.html)? I noticed the same issue in several other places too.

Anonymxtrix commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,27 @@
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String inData = "yo";
```
Could you perhaps explain to me what this is supposed to be? I don't quite understand this part. 

EDIT: Ahh... I get it now... But would it be more accurate to change it to a final variable `START_COMMAND` instead? Since it is passed into a method `executeCommand`.

Anonymxtrix commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,26 @@
+
+    @Override
+    public String getTask() {
+        return "E";
```
Any reason why you did it this way instead of using a constant or final variable? Same for a few other places...

Anonymxtrix commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,36 @@
+        return this.description;
+    }
+
+    public String getTask() {
```
Would it be more accurate to have named this `getTaskTypeIcon` following the style from `getStatusIcon` in line 12?

Anonymxtrix commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,289 @@
+
+    public void executeCommand(String inData) {
+        int inDataLength = inData.length();
+        if (!Objects.equals(inData.toLowerCase(), "bye")) {
```
Perhaps `!inData.toLowerCase().equals("bye")` would read more easily here?

Anonymxtrix commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,289 @@
+        if (!Objects.equals(inData.toLowerCase(), "bye")) {
+            if (Objects.equals(inData, "list")) {
+                listAll();
+            } else if (inData.contains("todo")) {
```
Nice! I like the way `inData.contains("todo")` reads as "In data contains todo" in english! Very comfortable to read 👍

Anonymxtrix reviewed (commented):
Nicely done! I really am able to see the amount of effort you put in to follow the coding standards [here](https://se-education.org/guides/guidelines/PRs-reviewing.html).

itsraveen commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,289 @@
+        }
+    }
+
+    public void executeCommand(String inData) {
```
I think a better way to implement this would be to extract all the methods for the different if else blocks instead of trying to handle them all in this executeCommand method. That'll help make the code easier to read and understand, while keeping each method simple and concise.

itsraveen commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,289 @@
+                        + "____________________________________________________________");
+    }
+
+    public void deadlineAddedMessage() {
```
I like how each of these are stored as methods, makes it easier to read in the code above. 👍

itsraveen commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,289 @@
+import java.util.Objects;
+
+public class TaskManager {
+    Task[] taskList = new Task[100];
```
Having an array size of 100 is fine here, but maybe can consider an ArrayList instead.

itsraveen reviewed (commented):
Code looks good with no major issues. Hope my suggestions are useful.

alyssa-savier committed:
Merge remote-tracking branch 'origin/add-gradle-support'
To enable gradle support to Duke/Joker project

alyssa-savier committed:
Add GUI for Duke

alyssa-savier committed:
Merge branch 'branch-Level-10'

alyssa-savier committed:
Integrate GUI with Duke Bot

alyssa-savier committed:
Implement more OOP

alyssa-savier committed:
Implement more OOP

alyssa-savier committed:
Implement JavaDocs and OOP

alyssa-savier committed:
Add JAvaDocs and Implement OOP

alyssa-savier committed:
Implement Coding Quality Improvements

Code quality improved by removing redundant variables.
This makes the code more readable, and does not slow down the code
unnecessarily.

It is important to have better quality code for better understanding.

alyssa-savier committed:
Improve Code Quality and Add Full Commit Message

Code quality improved by removing redundant variables.
This makes the code more readable, and does not slow down the code
unnecessarily.

It is important to have better quality code for better understanding.

alyssa-savier committed:
Improve coding standard

alyssa-savier committed:
Merge pull request #2 from alyssa-savier/branch-A-CodingStandard

Improve coding standard

alyssa-savier committed:
Add More JUnit Testcases

alyssa-savier committed:
Clean up code

alyssa-savier committed:
Add Assertions

alyssa-savier committed:
Merge branch 'branch-A-Assertions'

alyssa-savier committed:
Add Intro Message and Joker profile picture

alyssa-savier committed:
Add Help Command

alyssa-savier committed:
Add Ui screenshot

alyssa-savier committed:
Add User Guide

alyssa-savier committed:
Set theme jekyll-theme-tactile

alyssa-savier committed:
Implement bug fixes

alyssa-savier committed:
Merge branch 'master' of https://github.com/alyssa-savier/ip-1

alyssa-savier committed:
Edit OOP and nested methods

alyssa-savier committed:
Fix Find Command
